### PR TITLE
MultimapSerializer works with non-String keys

### PR DESCRIPTION
**Pull request by sjcorbett**
MultimapSerializer works with non-String keys

### CODE REVIEW

**Comment by sjcorbett** [comment]
Thanks @geomacy

File: core/src/test/java/org/apache/brooklyn/util/core/json/MultimapSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Test
+    public void testSerializeEntityKey() throws Exception {
+        Entity entity = app.createAndManageChild(EntitySpec.create(BasicEntity.class));
+        Multimap<Entity, Integer> map = MultimapBuilder.hashKeys().arrayListValues().build();
+        map.put(entity, 1);
+        map.put(entity, 2);
+        map.put(entity, 3);
+        String json = mapper.writer().writeValueAsString(map);
+        assertEquals(json, "{\"BasicEntityImpl{id=" + entity.getId() + "}" + "\":[1,2,3]}");
```
Comment (neykov) [inline core/src/test/java/org/apache/brooklyn/util/core/json/MultimapSerializerTest.java]: What would the output be if you tried to serialise a complex object, say `map.put(new SimpleBean(), 1)`?

Comment (sjcorbett) [inline core/src/test/java/org/apache/brooklyn/util/core/json/MultimapSerializerTest.java]: It should be the `toString` of that object. A SimpleBean would be something like `"SimpleBean@4e5cdbfd"`. A Multimap with an `ImmutableMap` key would look like: `{"{a=b}":[3]}`

Comment (sjcorbett) [inline core/src/test/java/org/apache/brooklyn/util/core/json/MultimapSerializerTest.java]: Assuming you mean SimpleBean in SoftwareProcessShellEnvironmentTest it'd actually be `"org.apache.brooklyn.entity.software.base.SoftwareProcessShellEnvironmentTest$SimpleBean@xxxxxxxx"`.

Comment (neykov) [inline core/src/test/java/org/apache/brooklyn/util/core/json/MultimapSerializerTest.java]: Thanks. I meant a generic `SimpleBean` :).

**Pull request merged**